src/utilities/__tests__/stripIgnoredCharacters-test.js

```diff
@@ -0,0 +1,489 @@
+          `\tto equal ${inspectStr(stripped)}\n` +
+          `\tbut got  ${inspectStr(strippedTwice)}`,
+      );
+    },
+    toStayTheSame() {
+      this.toEqual(docString);
+    },
+    toThrow(expectedStringifyError) {
+      let catchedError;
+
+      try {
+        stripIgnoredCharacters(docString);
+      } catch (e) {
+        catchedError = e;
+      }
+      expect(String(catchedError)).to.equal(expectedStringifyError);
+    },
+  };
+
+  function inspectStr(str) {
+    // Called only to make error messages for failing tests
+    /* istanbul ignore next */
+    return JSON.stringify(str)
+      .replace(/^"|"$/g, '`')
+      .replace(/\\"/g, '"');
+  }
+}
+
+describe('stripIgnoredCharacters', () => {
+  it('asserts that a source to parse was provided', () => {
+    // $DisableFlowOnNegativeTest
+    expect(() => stripIgnoredCharacters()).to.throw(
+      'Must provide string or Source. Received: undefined',
+    );
+  });
+
+  it('asserts that an invalid source to parse was provided', () => {
+    // $DisableFlowOnNegativeTest
+    expect(() => stripIgnoredCharacters({})).to.throw(
+      'Must provide string or Source. Received: {}',
+    );
+  });
+
+  it('strips ignored characters from GraphQL query document', () => {
+    const query = dedent`
+      query SomeQuery($foo: String!, $bar: String) {
+        someField(foo: $foo, bar: $bar) {
+          a
+          b {
+            c
+            d
+          }
+        }
+      }
+    `;
+
+    expect(stripIgnoredCharacters(query)).to.equal(
+      'query SomeQuery($foo:String!$bar:String){someField(foo:$foo bar:$bar){a b{c d}}}',
+    );
+  });
+
+  it('strips ignored characters from GraphQL SDL document', () => {
+    const sdl = dedent`
+      """
+      Type description
+      """
+      type Foo {
+        """
+        Field description
+        """
+        bar: String
+      }
+    `;
+
+    expect(stripIgnoredCharacters(sdl)).to.equal(
+      '"""Type description""" type Foo{"""Field description""" bar:String}',
+    );
+  });
+
+  it('report document with invalid token', () => {
+    expectStripped('{ foo(arg: "\n"').toThrow(dedent`
+      Syntax Error: Unterminated string.
+
+      GraphQL request (1:13)
+      1: { foo(arg: "
+                     ^
+      2: "
+    `);
+  });
+
+  it('strips non-parsable document', () => {
+    expectStripped('{ foo(arg: "str"').toEqual('{foo(arg:"str"');
+  });
+
+  it('strips documents with only ignored characters', () => {
+    expectStripped('\n').toEqual('');
+    expectStripped(',').toEqual('');
+    expectStripped(',,').toEqual('');
+    expectStripped('#comment\n, \n').toEqual('');
+
+    for (const ignored of ignoredTokens) {
+      expectStripped(ignored).toEqual('');
+
+      for (const anotherIgnored of ignoredTokens) {
+        expectStripped(ignored + anotherIgnored).toEqual('');
+      }
+    }
+
+    expectStripped(ignoredTokens.join('')).toEqual('');
```

Comment:
I wrote tests as a combination of handwritten test that covers algorithm and all branches and fuzzing tests that test all possible combinations of tokens.
Fuzzing tests helped to uncover a few edge cases that I fixed and include in handwritten tests but I think fuzzing tests are still useful in case if we change the code or extend GraphQL language grammar.